packages/eslint-plugin-template/src/rules/button-has-type.ts

```diff
@@ -0,0 +1,126 @@
+import type {
+  ParseSourceSpan,
+  TmplAstElement,
+} from '@angular-eslint/bundled-angular-compiler';
+import {
+  ASTWithSource,
+  LiteralPrimitive,
+} from '@angular-eslint/bundled-angular-compiler';
+import type {
+  BoundAttribute,
+  TextAttribute,
+} from '@angular/compiler/src/render3/r3_ast';
+import {
+  createESLintRule,
+  getTemplateParserServices,
+} from '../utils/create-eslint-rule';
+
+export const RULE_NAME = 'button-has-type';
+export const INVALID_TYPE_DATA_KEY = 'type';
+export type MessageIds = 'invalidType' | 'missingType';
+
+interface InvalidButtonTypeInfo {
```

Comment:
I really don't think these comments add any value here (especially the source span) and they can be removed